anesthetic/plot.py

```diff
@@ -47,10 +58,85 @@ def _constructor(self):
     def _constructor_expanddim(self):
         return AxesDataFrame
 
+    @staticmethod
+    def axes_series(data, fig, ncol=None, gridspec_kw=None, subplot_spec=None):
+        """Set up subplots for `AxesSeries`."""
+        axes = data.copy()
+        if fig is None:
+            fig = plt.figure()
+        if ncol is None:
+            ncol = int(np.ceil(np.sqrt(axes.index.size)))
+        ncol = ncol
+        nrow = int(np.ceil(axes.index.size / ncol))
+        if gridspec_kw is None:
+            gridspec_kw = {}
+        wspace = gridspec_kw.pop('wspace', 0)
+        if subplot_spec is None:
+            gs = GridSpec(nrow, ncol, wspace=wspace, **gridspec_kw)
+        else:
+            gs = GridSpecFromSubplotSpec(nrow, ncol, wspace=wspace,
+                                         subplot_spec=subplot_spec,
+                                         **gridspec_kw)
+        for p, g in zip(axes.index, gs):
+            axes[p] = ax = fig.add_subplot(g)
+            ax.set_yticks([])
+        return axes
+
+    @staticmethod
+    def _set_xlabels(axes, labels, **kwargs):
+        if labels is None:
+            labels = {}
+        labels = {p: labels[p] if p in labels else p for p in axes.index}
+        for p, ax in axes.iteritems():
+            ax.set_xlabel(labels[p], **kwargs)
```

Comment:
I want the labels to be set already upon initialisation. For that I need a staticmethod, because there cannot be any slicing happening in the `__init__`, meaning any sort of `self[p]` or `self.iloc[0]` or `self.iteritems()` cannot be put into `__init__`. Otherwise this messes with `_constructor` and `_constructor_expanddim` and `_constructor_sliced`.